File: electro-server/src/main/java/com/electro/entity/inventory/VariantInventory.java
package com.electro.entity.inventory;

import com.electro.entity.product.Variant;
import lombok.Data;

import java.util.List;

@Data
public class VariantInventory {
    private Variant variant;
    private List<DocketVariant> transactions;
    private Integer actualInventory;
    private Integer waitingForDelivery;
    private Integer canBeSold;
    private Integer areComing;
}

File: electro-client/src/models/VariantInventory.ts
import BaseResponse from 'models/BaseResponse';
import { CollectionWrapper } from 'types';
import { VariantPropertyItem } from 'models/Variant';
import { DocketVariantEliminatedResponse } from 'models/DocketVariantEliminatedResponse';

export interface VariantInventoryResponse {
  variant: VariantResponse;
  transactions: DocketVariantEliminatedResponse[];
  inventory: number;
  waitingForDelivery: number;
  canBeSold: number;
  areComing: number;
}

interface VariantResponse extends BaseResponse {
  product: ProductResponse;
  sku: string;
  cost: number;
  price: number;
  properties: CollectionWrapper<VariantPropertyItem> | null;
  status: number;
}

interface ProductResponse extends BaseResponse {
  name: string;
  code: string;
  slug: string;
}
